gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -491,7 +491,7 @@ public void runJob(Properties jobProps, JobListener jobListener) throws JobExcep
     try {
       Spec flowSpec = this.scheduledFlowSpecs.get(jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY));
       String triggerTimestampMillis = jobProps.getProperty(
-          ConfigurationKeys.ORCHESTRATOR_TRIGGER_EVENT_TIME_MILLIS_KEY, "0");
+          ConfigurationKeys.ORCHESTRATOR_TRIGGER_EVENT_TIME_MILLIS_KEY, "-1");
```

Comment:
if this should "never happen", maybe a comment just above, expressing that it should always be set... and if ever not, the orchestrator will fail the flow.